ui/utils.py

```diff
@@ -170,3 +177,119 @@ def write_x509_files():
     """Write the x509 certificate and key to files"""
     write_to_file(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_CERTIFICATE)
     write_to_file(settings.MIT_WS_PRIVATE_KEY_FILE, settings.MIT_WS_PRIVATE_KEY)
+
+
+def get_video_analytics(video_key):
+    """Get video analytics data from Google Analytics."""
+    ga_client = get_google_analytics_client()
+    ga_response = ga_client.reports().batchGet(
+        body=generate_google_analytics_query(video_key)).execute()
+    try:
+        return parse_google_analytics_response(ga_response)
+    except Exception as exc:
+        raise GoogleAnalyticsException(
+            'Could not parse analytics response') from exc
+
+
+def get_google_analytics_client():
+    """Gets a Google Analytics client.
+
+    Returns:
+        analytics_client: An analytics client
+    """
+    _check_files_exist([settings.GA_KEYFILE])
+    try:
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            settings.GA_KEYFILE)
+        analytics_client = build('analyticsreporting', 'v4',
+                                 credentials=credentials)
+        return analytics_client
+    except Exception as exc:  # pylint: disable=broad-except
+        raise GoogleAnalyticsException('Something went wrong with creating a'
+                                       'GoogleAnaltics client') from exc
+
+
+def generate_google_analytics_query(video_key):
+    """Generates a Google Analytics query.
+
+    Returns:
+        analytics_query: a query dict suitable to use as the body of an
+        analytics 'batchGet' request.
+    """
+    if not settings.GA_VIEW_ID:
+        raise RuntimeError("Missing 'settings.GA_VIEW_ID'.")
+    ga_view_id = settings.GA_VIEW_ID
+    # https://developers.google.com/analytics/devguides/reporting/core/v3/reference
+    START_DATE = '2005-01-01'
+    END_DATE = '9999-01-01'
+    query = {
+        'reportRequests': [
+            {
+                'viewId': ga_view_id,
+                'dateRanges': [{'startDate': START_DATE, 'endDate': END_DATE}],
+                'metrics': [{'expression': 'ga:totalEvents'}],
+                'dimensions': [
+                    # [adorsk, 2018-03]
+                    # Achtung! a high degree of implicit coupling to
+                    # dimension names that have been manually set in GA.
+                    # Hard-coding for now.
+                    {'name': 'ga:eventAction'},
+                    {'name': 'ga:dimension1'}
+                ],
+                'dimensionFilterClauses': [
+                    {
+                        'filters': [
+                            {
+                                'dimensionName': 'ga:eventLabel',
+                                'operator': 'EXACT',
+                                'expressions': [video_key]
```

Comment:
Good catch on the capitalization issue.  While adding `titleCase: false` to the react-ga initialization will stop the video keys from being capitalized, it will also cause a split in the analytics data for the videos that begin with a letter (before and after fix).  Another possible option is to use `video_key.capitalize()` here, which should avoid that data split.   Might be worth getting input from @pdpinch on the preferred approach.

Reply:
re: @mbertrand's note about splitting analytics data.

To me, the tradeoff here is between these things:
  
    1. CHANGE FRONT-END: We change our front-end code so that ga data is not altered.
        PROS:
            1. Future analytics data matches what we expect.
        CONS:
            1. We have to handle events that happen before this change. some options here:
                1. ignore 'legacy' events 
                2. change the backend to search for both new + legacy events (easy, just change backend GA query)
    2. DON'T CHANGE FRONT-END: We leave front-end code as-is.
        a. PROS:
            1. analytics data is consistent (no legacy vs new formats)
        b. CONS:
            1. We have to change the backend to search for legacy events (easy)
            2. If we add future analytics queries, we must remeber that event data is alteread by react-ga.


I personally favor <CHANGE FRONT-END> and ignoring legacy events, since our earliest events only start in 2017-10:
https://analytics.google.com/analytics/web/#report/content-event-overview/a5145472w162315911p163330947/%3F_u.date00%3D20150314%26_u.date01%3D20180320/

@pdpinch, whaddya think?

Reply:
I don't think we put a lot of care into deciding whether the events should be `t...` or `T...` so I think consistent data from Google is more valuable than lower-case t. 

That said, I would like to have the front-end modified so that it will be obvious to the next engineer what to expect. Can we specify that the events be upper-case? 

Does that make sense?

Reply:
Roger that, sticking with capitalized events.
I'll add a comments which tell future engineers that events get capitalized.